src/pipelines/real-estate/realestate/common/solids_spark_delta.py

```diff
@@ -295,54 +296,58 @@ def _sql_solid(context, **input_defs):  # pylint: disable=unused-argument
             DeltaCoordinate:
                 The Delta Table Coordinates where the SQL statements were running against.
         """
-        if input_defs['target_delta_table'] is None:
+        if ins["target_delta_table"] is None:
             raise Exception("Input `target_delta_table` not provided.")
-        if input_defs['input_dataframe'] is None:
+        if ins["input_dataframe"] is None:
             raise Exception("Input `input_dataframe` not provided.")
         ##
         ## Handling delta-table
         ##
         target_delta_path = _get_s3a_path(
-            input_defs['target_delta_table']["s3_coordinate_bucket"],
-            input_defs['target_delta_table']["s3_coordinate_key"],
+            ins["target_delta_table"]["s3_coordinate_bucket"],
+            ins["target_delta_table"]["s3_coordinate_key"],
         )
         context.log.info("Target Delta table path: {}".format(target_delta_path))
 
         # prepare colums for merge statement
-        insert_columns = '\n, '.join(input_defs['input_dataframe'].columns)
-        update_columns = '\n, '.join(
-            ['trg.' + c + ' = src.' + c for c in input_defs['input_dataframe'].columns]
+        insert_columns = "\n, ".join(ins["input_dataframe"].columns)
+        update_columns = "\n, ".join(
+            ["trg." + c + " = src." + c for c in ins["input_dataframe"].columns]
         )
 
         # Set Delta-table path and columns
         sql_statement_template = Template(sql_statement)
         repl_sql_statement = sql_statement_template.render(
-            target_delta_table='delta.`' + target_delta_path + '`',
+            target_delta_table="delta.`" + target_delta_path + "`",
             update_columns=update_columns,
             insert_columns=insert_columns,
         )
 
         context.log.info(
-            "Executing sql statement:\n{sql_statement}".format(sql_statement=repl_sql_statement)
+            "Executing sql statement:\n{sql_statement}".format(
+                sql_statement=repl_sql_statement
+            )
         )
 
         ##
         ## Handling spark dataframe
         ##
 
         # register input df for spark to be available in spark.sql
-        input_defs['input_dataframe'].createOrReplaceTempView("input_dataframe")
+        ins["input_dataframe"].createOrReplaceTempView("input_dataframe")
 
         context.resources.pyspark.spark_session.sql(
```

Comment:
Try to find alternatives to spark. E.g. using DuckDB on top of S3 or delta-rs direct.